solrsearch-plugins/src/test/java/co.cask.hydrator.plugin/SolrSearchSinkTest.java

```diff
@@ -0,0 +1,291 @@
+                                                                              CURRENT_VERSION, true,
+                                                                              CURRENT_VERSION, true);
+  private static final String SINGLE_NODE_SOLR_MODE = "Single Node Solr";
+  private static final String SOLR_CLOUD_MODE = "SolrCloud";
+  @ClassRule
+  public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+    setupBatchArtifacts(BATCH_APP_ARTIFACT_ID, ETLBatchApplication.class);
+
+    setupRealtimeArtifacts(REALTIME_APP_ARTIFACT_ID, ETLRealtimeApplication.class);
+
+    Set<ArtifactRange> parents = ImmutableSet.of(BATCH_ARTIFACT_RANGE, REALTIME_ARTIFACT_RANGE);
+
+    // add Solr search plugins
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("solrsearch-plugins", "1.0.0"), parents,
+                      SolrSearchSink.class, SolrSearchSinkConfig.class, RealtimeSolrSearchSink.class);
+  }
+
+  @Ignore
+  public void testBatchSolrSearchSink() throws Exception {
+    String inputDatasetName = "solr-batch-input-source";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputDatasetName));
+
+    Map<String, String> sinkConfigproperties = new ImmutableMap.Builder<String, String>()
+      .put(Constants.Reference.REFERENCE_NAME, "BatchSolrSink")
+      .put("solrMode", SINGLE_NODE_SOLR_MODE)
+      .put("solrHost", "localhost:8983")
+      .put("collectionName", "collection1")
+      .put("idField", "rowid")
+      .put("outputFieldMappings", "rowid:id")
+      .build();
+
+    ETLStage sink = new ETLStage("SolrSink", new ETLPlugin("SolrSearch", BatchSink.PLUGIN_TYPE, sinkConfigproperties,
+                                                           null));
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    ApplicationId appId = NamespaceId.DEFAULT.app("testBatchSolrSink");
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(inputSchema).set("rowid", "1").set("firstname", "Brett").set("lastname", "Lee").set
+        ("address", "NE lake side").set("pincode", 480001).build(),
+      StructuredRecord.builder(inputSchema).set("rowid", "2").set("firstname", "John").set("lastname", "Ray").set
+        ("address", "SE lake side").set("pincode", 480002).build()
+    );
+    MockSource.writeInput(inputManager, input);
+
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
```

Comment:
I think you should validate the data in solr, similar to how you do in the realtime test cse